api/api.go

```diff
@@ -292,3 +292,44 @@ func (c *Client) Skip(language, slug string) error {
 
 	return errors.New(pe.Error)
 }
+
+// Status sends a request to exercism to fetch the user's
+// completion status for the given language track.
+func (c *Client) Status(language string) ([][]string, error) {
+	url := fmt.Sprintf("%s/tracks/%s/status?key=%s", c.XAPIHost, language, c.APIKey)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.Do(req, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusNoContent {
+		msg := fmt.Sprintf("You have not yet begun the %v track", language)
+		return nil, errors.New(msg)
+	}
+	//
+	//
+	//
+	//
+	//
+	// API response format
+	// {
+	//   "recent":"leap",
+	//   "submitted":"Jan 1, 2015",
```

Comment:
I think we should have the track id in the response for completeness, even though you have to know the language in order to make the API call.

Also, let's make "submitted" a timestamp called "submitted_at", and use the iso8601 format for it. The API shouldn't have to know when we want timestamps vs when we just want the date.

Lastly, I think we should put the recent/submitted together so that it's clear what the submitted timestamp means:

```
{
  "track_id": "go",
  "recent": {
    "problem": "leap",
    "submitted_at": "2015-01-01T14:31:57Z"
  },
  "fetched": ["gigasecond", "raindrops"],
  "skipped": ["clock", "poker"]
}
```